### PR TITLE
Update Solr search handler to implement upcoming changes in eZ\Publish\SPI\Search\Handler interface

### PR DESCRIPTION
**Pull request by alongosz**
This PR is related to #48 and JIRA issues: [EZP-25088](https://jira.ez.no/browse/EZP-25088) and [EZP-25756](https://jira.ez.no/browse/EZP-25756).

The main goal is to avoid CI issues in dependencies when merging [changes](https://github.com/ezsystems/ezpublish-kernel/pull/1660/files) to `eZ\Publish\SPI\Search\Handler` interface.

It includes:
- [x] Implementation of new methods `EzSystems\EzPlatformSolrSearchEngine\Handler::generateDocument` and `EzSystems\EzPlatformSolrSearchEngine\Handler::bulkIndexDocuments` required by upcoming version of `eZ\Publish\SPI\Search\Handler` interface.
- [x] Better logging of database inconsistencies while executing `ezplatform:solr_create_index` .

### CODE REVIEW

**Comment by alongosz** [comment]
Ready for review.
Ping @andrerom

**Comment by andrerom** [comment]
+1

File: bundle/Command/SolrCreateIndexCommand.php [hunk 1]
```diff
@@ -94,15 +95,14 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
-                    $output->write("\r"); // get rid of padding (side effect of displaying progress bar)
-                    $logger->warning("Could not load current version of Content with id ${row['id']}, so skipped for indexing. Full exception: " . $e->getMessage());
-                    $progress->display();
+                    $this->logWarning($output, $progress, "Could not load current version of Content with id ${row['id']}, so skipped for indexing. Full exception: " . $e->getMessage());
                 }
             }
 
-            if (!empty($contentObjects)) {
-                $searchHandler->bulkIndexContent($contentObjects);
+            $documents = $this->generateDocuments($contentObjects, $searchHandler, $progress, $output);
```
Comment (andrerom) [inline bundle/Command/SolrCreateIndexCommand.php]: *could* rename this to _not_ be same name as on handler given this does more, like `generateDocumentsWithOuput` or something.

or just inline it, in current form method is not reusable anyway.

File: bundle/Command/SolrCreateIndexCommand.php [hunk 2]
```diff
@@ -113,4 +113,49 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+     */
+    private function generateDocuments(array $contentObjects, SolrSearchEngineHandler $searchHandler, ProgressHelper $progress, OutputInterface $output)
+    {
+        $documents = [];
+
+        foreach ($contentObjects as $content) {
+            try {
+                $documents[] = $searchHandler->generateDocument($content);
+            } catch (NotFoundException $e) {
+                // ignore content objects without assigned state id
```
Comment (andrerom) [inline bundle/Command/SolrCreateIndexCommand.php]: comment is invalid, NotFound can be several things, so:

`// Ignore content objects that have some sort of missing data on it`

File: bundle/Command/SolrCreateIndexCommand.php (hunk 1 above)
Comment (alongosz) [inline bundle/Command/SolrCreateIndexCommand.php]: Ok, it's inline now

File: bundle/Command/SolrCreateIndexCommand.php (hunk 2 above)
Comment (alongosz) [inline bundle/Command/SolrCreateIndexCommand.php]: Changed, also changed logged message to reflect this.

**Comment by alongosz** [comment]
@andrerom: fixed as requested in comments :)

**Comment by andrerom** [comment]
Thanks, merging and tagging, so this will be picked up by your kernel PR.

**Pull request merged**